src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -205,10 +207,12 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
 					if (!StringUtils.isNotBlank(property.getBuildRunPolicy())) {
 						property.setBuildRunPolicy(buildRunPolicy);
 					}
+					property.setSuccessfulBuildsHistoryLimit(successfulBuildsHistoryLimit);
+					property.setFailedBuildsHistoryLimit(failedBuildsHistoryLimit);
 					return property;
 				} else {
 					return new BuildConfigProjectProperty(namespace, buildConfigName, uuid, resourceVersion,
-							buildRunPolicy);
+							buildRunPolicy, successfulBuildsHistoryLimit, failedBuildsHistoryLimit);
```

Comment:
so every time we add a new field to the BuildConfig api, we need to update the sync plugin to locally store that field value?  That's going to cause us a lot of api compatibility/migration pain.  Seems like we need a more generic solution that doesn't drop fields it doesn't know about.

Reply:
That is correct.  A more generic solution would be great, should we create a new card for it?

Reply:
yes. we should probably be using patch to update the buildconfig instead of update.  that way fields we don't know about don't get touched.

Reply:
From what I can tell, we are building a new config and doing a replace on line #272-274

Reply:
Also, this for loop never actually loops!  Both paths call return, should I go ahead and fix that now?

Reply:
well it should be fixed but i'm not sure what the fix is.  i'm not sure what the intent was in trying to create the buildconfigproject property in all the namespaces the sync plugin is watching.  If it doesn't create it in all of them, which one should it be creating it in?

But yes, this seems broken as it currently stands (and probably fails for anyone using the sync plugin against multiple namespaces)

Reply:
These original fabric guys liked their 2-way sync ;-) ... this is how I interpret the path ... the item listener sees an update to the job, and then triggers an bc update, trying to correlate any particulars.  For example, if they edited an embedded Jenkinsfile pipeline from the jenkins console after it was imported from a BC.

For what is worth, not to imply there isn't something to fix here, but to date, there have been no bugzillas/issues filed that caught the anomalies you all have mentioned.

Maybe we are just lucky and @coreydaley discovered it before others.

Reply:
@coreydaley could you opens a sync plugin issue to capture this so we can track it ... perhaps it is a complex enough bug that it rolls into a card, but let's start with a github issue

Reply:
https://github.com/openshift/jenkins-sync-plugin/issues/200